src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -108,9 +109,13 @@ public AzureMonitorClient(AzureEnvironment azureCloud, string tenantId, string s
 
                 // Get the metric value according to the requested aggregation type
                 var requestedMetricAggregate = InterpretMetricValue(aggregationType, mostRecentMetricValue);
-
-                var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
-                measuredMetrics.Add(measuredMetric);
+                try {
+                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
+                    measuredMetrics.Add(measuredMetric);
+                } catch (ArgumentException) {
+                    _logger.LogWarning("{MetricName} has return a time series with empty value for {Dimension} and the measurements will be dropped", metricName, metricDimension); 
+                    _logger.LogDebug("The violating time series has content {}", JsonConvert.SerializeObject(timeseries)); 
+                }
```

Comment:
Let's align with .NET style:
```suggestion
                try
                {
                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
                    measuredMetrics.Add(measuredMetric);
                }
                catch (ArgumentException)
                {
                    _logger.LogWarning("{MetricName} has return a time series with empty value for {Dimension} and the measurements will be dropped", metricName, metricDimension); 
                    _logger.LogDebug("The violating time series has content {}", JsonConvert.SerializeObject(timeseries)); 
                }
```